File: C/G2/G2.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool una_b_mas_que_a(const char *s) {
    int a = 0, b = 0;
    for (const char *p = s; *p; p++) {
        if (*p == 'a') a++;
        else if (*p == 'b') b++;
        else return false;
    }
    return b == a + 1;
}

void chomp(char *s) {
    int n = strlen(s);
    while (n > 0 && (s[n-1] == '\n' || s[n-1] == '\r')) {
        s[n-1] = '\0';
        n--;
    }
}

int main() {
    FILE *f = fopen("G2.txt", "r");
    if (!f) { printf("[ADVERTENCIA] No se encontró G2.txt\n"); return 0; }
    char linea[256];
    int i = 0;
    while (fgets(linea, sizeof(linea), f)) {
        chomp(linea);
        i++;
        bool acepta = una_b_mas_que_a(linea);
        printf("[G2][%d] %s => %s\n", i, linea[0] ? linea : "ε", acepta ? "acepta" : "NO acepta");
    }
    fclose(f);
    return 0;
}

File: Python/G2/G2.py
from pathlib import Path

def una_b_mas_que_a(s: str) -> bool:
    if not all(c in "ab" for c in s):
        return False
    return s.count('b') == s.count('a') + 1

def main():
    archivo = Path("G2.txt")
    if not archivo.exists():
        print("[ADVERTENCIA] No se encontró G2.txt")
        return
    with archivo.open("r", encoding="utf-8") as f:
        for i, linea in enumerate(f, start=1):
            s = linea.strip()
            acepta = una_b_mas_que_a(s)
            veredicto = "acepta" if acepta else "NO acepta"
            mostrar = s if s else "ε"
            print(f"[G2][{i}] {mostrar} => {veredicto}")


main()